### PR TITLE
Ft signup validation #184638231

### PR DESCRIPTION
**Pull request by nshimiyeemmy**
What does this PR do?

This PR adds request validation to the request body before registering a user. It validates that all required fields are set and are in acceptable format and within a specified character range.

Description of Task to be completed?

Create a user schema for an acceptable data format to register a new user
Validated request body against the Shema in auth.controller
Return detailed error message when validation fails

How should this be manually tested?

Start your local development server
Open postman and send a POST request to localhost:3000/api/v1/auth/register
set request body type to JSON
use a request body that does not include all required fields or a password of 3 characters or a malformed email

Relevant Pivotal Tracker task

Users should receive descriptive signup/Registration validation errors

Task with ID: #184638231

### CODE REVIEW

File: src/types/token.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export interface Tokens {
```
[the remark this answers is not in the document]
Comment (nshimiyeemmy) [reply src/types/token.ts:1]: DONE. allowed ECMA script modules in eslint config

**Review by iradukundairenee — CHANGES_REQUESTED** [review]
please fix the error in the github actions.

**Review by gershomlapaix — COMMENTED** [review]
ready

**Comment by iradukundairenee** [comment]
> ready

@gershomlapaix  do you mean this PR is read for being approved?

**Review by iradukundairenee — APPROVED** [review]

**Review by gershomlapaix — APPROVED** [review]

**Pull request merged**